src/infra/flow-system-inmem/tests/tests/test_flow_service_inmem.rs

```diff
@@ -1287,6 +1297,39 @@ impl FlowHarness {
         }
         task_states
     }
+
+    fn now_datetime(&self) -> DateTime<Utc> {
+        self.system_time_source_stub.now()
+    }
+
+    async fn simulate_time_passage(&self, time_quantum: Duration) {
+        // Examples:
+        // 12 ÷ 4 = 3
+        // 15 ÷ 4 = 4
+        // 16 ÷ 4 = 4
+        fn div_up(a: i64, b: i64) -> i64 {
+            (a + (b - 1)) / b
+        }
+
+        const TIME_INCREMENT: Duration = Duration::milliseconds(SCHEDULING_ALIGNMENT_MS);
+        const CONTEXT_SWITCHING_SLEEP_DURATION: StdDuration = if cfg!(windows) {
+            // Windows sleep() is accurate up to about ~20-50 milliseconds
+            StdDuration::from_millis(50)
+        } else {
+            StdDuration::from_millis(5)
```

Comment:
Does it matter for faked time at all?

Reply:
yes, we need pauses so that asychronous code has time to work (context switching)

also, when we have multiple calls for a very short period of time, we cannot bypass them with a single `sleep` call

Reply:
I don't get it. These are tests with faked time, not real time. Why do we care about real time issues on Windows ?

Reply:
Oh, I see -- you are right, that was extra things -- replaces with `yield_now()` (we definitely don't need to sleep here at all)